### ISSUE TITLE
How to visualize intermediate activations with tf-hub?

### ISSUE THREAD

**Issue by rgnespolo** [tensorflow/hub#569]
Hi!

I am using tf-hub for transfer learning in image classification problems.
When I import the model (https://tfhub.dev/google/tf2-preview/mobilenet_v2/feature_vector/4) and add a few more dense layers, I cannot get access easily to the intermediate layers to plot the activation  results of the images (suck as made with keract, since I end up getting only the feature vector 1D values).

Has anyone ever implemented this visualization within the internal layers of tf-hub models?
Below, my model.

Thank you!

`

> model = tf.keras.Sequential([
> hub.KerasLayer("https://tfhub.dev/google/tf2-preview/mobilenet_v2/feature_vector/4",input_shape=(224,224,3),
> output_shape=[1280],
> trainable=False),
> tf.keras.layers.Dense(64, activation='relu', name= 'convout1'),
> tf.keras.layers.Dropout(0.8),
> tf.keras.layers.Dense(train_generator.num_classes, activation='softmax')
> ])
> 
> 
> 
> model.build([None, 224, 224, 3])
> model.summary()
> model.compile(
> optimizer = tf.keras.optimizers.Adam(learning_rate=0.00002, beta_1=0.9, beta_2=0.999, amsgrad=False),
> loss='categorical_crossentropy',#categorical_crossentropy',#'mean_squared_error',
> metrics=['acc'])
> steps_per_epoch = np.ceil(train_generator.samples/train_generator.batch_size)

> val_steps_per_epoch = np.ceil(valid_generator.samples/valid_generator.batch_size)
> 
> from keras.callbacks import EarlyStopping
> early_stopping = EarlyStopping(monitor='val_loss', patience=2)
> 
> logdir = os.path.join("logs", datetime.datetime.now().strftime("%Y%m%d-%H%M%S"))
> tensorboard_callback = tf.keras.callbacks.TensorBoard(logdir, histogram_freq=1)
> 
> hist = model.fit(
> train_generator,
> epochs=30,
> verbose=1,
> steps_per_epoch=steps_per_epoch, 
> validation_data=valid_generator,
> callbacks=[tensorboard_callback],
> validation_steps=val_steps_per_epoch
> ).history
> 

`

**Comment by maringeo** [comment]
Hi @rgnespolo, in general intermediate layers are not exposed - https://github.com/tensorflow/hub/issues/453 provides more context and a workaround for TF2 SavedModels exported from TF-Slim.
Duplicate of #453
